src/realm/group.hpp

```diff
@@ -1002,7 +1002,7 @@ inline ref_type Group::get_history_ref(const Array& top) noexcept
 inline int Group::get_history_schema_version(const Array& top) noexcept
 {
     if (top.is_attached()) {
-        if (top.size() >= 8) {
+        if (top.size() >= 9) {
```

Comment:
This expression could also be extracted into a named bool. I see that it follows the pattern in this file, but we have a lot of number literals in this area. `get_history_ref` above has a check very similar to this.